crates/key-value-azure/src/store.rs

```diff
@@ -151,6 +160,150 @@ impl Store for AzureCosmosStore {
     async fn get_keys(&self) -> Result<Vec<String>, Error> {
         self.get_keys().await
     }
+
+    async fn get_many(&self, keys: Vec<String>) -> Result<Vec<(String, Option<Vec<u8>>)>, Error> {
+        let in_clause: String = keys
+            .into_iter()
+            .map(|k| format!("'{}'", k))
+            .collect::<Vec<String>>()
+            .join(", ");
+        let stmt = Query::new(format!("SELECT * FROM c WHERE c.id IN ({})", in_clause));
+        let query = self
+            .client
+            .query_documents(stmt)
+            .query_cross_partition(true);
+
+        let mut res = Vec::new();
+        let mut stream = query.into_stream::<Pair>();
+        while let Some(resp) = stream.next().await {
+            let resp = resp.map_err(log_error)?;
+            for (pair, _) in resp.results {
+                res.push((pair.id, Some(pair.value)));
+            }
+        }
+        Ok(res)
+    }
+
+    async fn set_many(&self, key_values: Vec<(String, Vec<u8>)>) -> Result<(), Error> {
+        for (key, value) in key_values {
+            self.set(key.as_ref(), &value).await?
+        }
+        Ok(())
+    }
+
+    async fn delete_many(&self, keys: Vec<String>) -> Result<(), Error> {
+        for key in keys {
+            self.delete(key.as_ref()).await?
+        }
+        Ok(())
+    }
+
+    async fn increment(&self, key: String, delta: i64) -> Result<i64, Error> {
+        let operations = vec![Operation::incr("/value", delta).map_err(log_error)?];
+        let _ = self
+            .client
+            .document_client(key.clone(), &key.as_str())
+            .map_err(log_error)?
+            .patch_document(operations)
+            .await
+            .map_err(log_error)?;
+        let pair = self.get_pair(key.as_ref()).await?;
+        match pair {
```

Comment:
I'm going to dig a little deeper. It wasn't clear initially that patch supported that.

Reply:
Looks like the REST API returns the original document and the Rust SDK throws it all away except the document attributes.

Reply:
@itowlson I think this would be the only one of me comments where we might consider blocking the PR on. What are your thoughts on whether the race here is worth blocking the PR?

Reply:
@rylev My preference is to merge as is.  We can document it as a known issue in the RC, continue investigating between now and release, and assuming no joy, we can either:

* Keep documenting it as a known issue in the Cosmos back end; or
* Change it to _always_ error with a "not supported on this back end" message

before release.